monkey/monkey_island/cc/repositories/plugin_archive_parser.py

```diff
@@ -26,27 +35,59 @@ def tarinfo_type(tar: TarInfo) -> str:
     return "unknown"
 
 
-def parse_plugin(file: BinaryIO) -> AgentPlugin:
+def parse_plugin(file: BinaryIO, data_dir: Path) -> Mapping[OperatingSystem, AgentPlugin]:
     """
     Load a plugin from a tar file.
 
     :raises ValueError: If the file is not a valid plugin
     """
+
+    parsed_plugin = {}
+
     tar_file = TarFile(fileobj=file)
     try:
         manifest = get_plugin_manifest(tar_file)
         schema = get_plugin_schema(tar_file)
         source = get_plugin_source(tar_file)
+
+        extracted_plugin = _safe_extract_file(tar=tar_file, filename="plugin.tar")
```

Comment:
Ok, so it seems that plugin tar archive must have a `plugin.tar` inside it and it needs to contain `plugin.py` as an entry point? Is the plugin structure documented somewhere? Wouldn't it make more sense to call this `source.tar` and have `main.py` in it as an entry point?

Reply:
> Is the plugin structure documented somewhere?

Not yet. It's not finalized and for now plugins are an "internal" interface.

> Wouldn't it make more sense to call this `source.tar`

This assumes that the plugin is composed of source code. We may allow compiled plugins in the future.

> and have `main.py` in it as an entry point?

`main` is the entry point into a program. A plugin is not a program.

Reply:
> Not yet. It's not finalized and for now plugins are an "internal" interface.
I would rather we did document it somewhere, even if it's an `.md` file. It would make it easier to migrate to an actual documentation and people wouldn't be compelled to leave comments like ` # if vendor/ exists, we don't want to check if vendor-linux/ or vendor-windows/ exist`

So far it looks like everything is named a plugin. manifest is `plugin.yaml`, the executable is also `plugin.tar`, the plugins themselves are called plugins, etc. I think the only thing that's a plugin is the whole `.tar` archive, everything else is a component of a plugin. At some point we should give proper names to those components, the sooner we do it the better

Reply:
I created https://github.com/guardicore/monkey/issues/2849. We can discuss specifics in planning on Thursday.